### ISSUE TITLE
NumberBox: Background of the SpinButton Popup cannot be distinguished from a default page background in Dark theme

### ISSUE THREAD

**Issue by Felix-Dev** [microsoft/microsoft-ui-xaml#2843]
**Describe the bug**
The background of the NumberBox SpinButton Popup cannot be distinguished from a default page background in Dark theme.

**Steps to reproduce the bug**
1. Open the XAML Controls Gallery.
2. Go to the NumberBox page and the specific NumberBox SpinButton example.
3. Make sure the NumberBox is in *compact* SpinButtonPlacementMode and focus the NumberBox.

![image](https://user-images.githubusercontent.com/1398851/86625242-0f0a0980-bfc5-11ea-8e54-18f86d59663f.png)

**Expected behavior**
Since the Popup is a transient UI like the ComboBox's dropdown menu or the MenuBarItem flyout menu, it should have the same background like these, as per the [acrylic guidance](https://docs.microsoft.com/en-us/windows/uwp/design/style/acrylic#when-to-use-acrylic):

| ComboBox | NumberBox |
|-------|------------|
|![image](https://user-images.githubusercontent.com/1398851/86625726-d74f9180-bfc5-11ea-84a2-1f51c94e73e6.png)|![image](https://user-images.githubusercontent.com/1398851/86631807-dbcc7800-bfce-11ea-9ef9-76246ea32b63.png)|

NuGet package version: 
Current WinUI master builds.

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| May 2020 Update (19041)            | Yes  |
| November 2019 Update (18363)       |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            | Yes |
| Xbox               |  |
| Surface Hub        |  |
| IoT                |  |

## Additional Context
If I were to work on a PR for this I would likely want to tie this into https://github.com/microsoft/microsoft-ui-xaml/issues/2844 as I would create a *dedicated* theme resource here for the Popup background.

**Comment by StephenLPeters** [comment]
@chigy can you weigh in on the appropriate color for this? @SavoySchuler FYI

**Comment by Felix-Dev** [comment]
For reference, I looked at the ComboBox theme resource file and just copied them over to the NumberBox:
https://github.com/microsoft/microsoft-ui-xaml/blob/f68b281afb086e37d62e3b39380108b10baa88ad/dev/ComboBox/ComboBox_themeresources.xaml#L75

Here are the resulting NumberBox theme resources then (*Dark* theme example)
```xml
<contract7NotPresent:StaticResource x:Key="NumberBoxPopupBackground" ResourceKey="SystemControlBackgroundChromeMediumLowBrush" />
            <contract7Present:StaticResource x:Key="NumberBoxPopupBackground" ResourceKey="SystemControlTransientBackgroundBrush" />
            <contract7NotPresent:StaticResource x:Key="NumberBoxPopupBorderBrush" ResourceKey="SystemControlForegroundChromeHighBrush" />
            <contract7Present:StaticResource x:Key="NumberBoxPopupBorderBrush" ResourceKey="SystemControlTransientBorderBrush" />
```

where `SystemControlTransientBackgroundBrush` maps to 
```xml
<AcrylicBrush x:Key="SystemControlTransientBackgroundBrush" BackgroundSource="HostBackdrop" TintColor="{StaticResource SystemChromeAltHighColor}" TintOpacity="0.8" FallbackColor="{StaticResource SystemChromeMediumLowColor}" />
```
and `SystemControlTransientBorderBrush` to
```xml
<SolidColorBrush x:Key="SystemControlTransientBorderBrush" Color="#000000" Opacity="0.36" />
```
in *Dark* theme.


| Theme | Current | Updated -1809+ (contract 7 present) | Updated -1803 and less (contract 7 not present) |
|-----|------|-----|----|
| Light | ![image](https://user-images.githubusercontent.com/1398851/86671330-d46c9500-bff5-11ea-922d-ea7236917c6e.png)|![image](https://user-images.githubusercontent.com/1398851/86671188-b2731280-bff5-11ea-8011-9d075c7de1cd.png) |![image](https://user-images.githubusercontent.com/1398851/86672194-b3587400-bff6-11ea-92e1-52097925275e.png)|
| Dark|![image](https://user-images.githubusercontent.com/1398851/86671410-e77f6500-bff5-11ea-91a7-716db787fbf3.png)| ![image](https://user-images.githubusercontent.com/1398851/86671075-94a5ad80-bff5-11ea-8b88-b0a36d95c25c.png)|![image](https://user-images.githubusercontent.com/1398851/86671951-72f8f600-bff6-11ea-86be-7d728f8c5a0e.png)|
  
I used the ComboBox dropdown theme resources here, as said above. I picked them for some initial experimenting here due to the similarity as both are a popup/flyout and thus their background appearance should probably match.

**Comment by mdtauk** [comment]
The shadow does not show up well with the pure black of the Dark Theme - so if this were to use Acrylic and borders, it would improve the experience.

**Comment by chigy** [comment]
@kikisaints , I agree with the suggestion but I want to make sure acrylic does not cause problem with this small surface. Kiki, is the size of this popup big enough not to cause the problem we had with Tooltip that made us decide not to use acrylic there?

**Comment by Felix-Dev** [comment]
@chigy As a helper, here's how the acrylic brush looks with colored background given this design:
![image](https://user-images.githubusercontent.com/1398851/86825038-362d0d80-c08f-11ea-8285-dc1eff7aa7c2.png)

| Light Theme | Dark Theme |
|----|-----|
|![image](https://user-images.githubusercontent.com/1398851/86825276-77252200-c08f-11ea-9f6e-eee1a26e63af.png)|![image](https://user-images.githubusercontent.com/1398851/86825080-42b16600-c08f-11ea-8105-a889d42dda68.png)|

**Comment by Felix-Dev** [comment]
@chigy Do you have an update on this?

**Comment by chigy** [comment]
@Felix-Dev , I was waiting for @kikisaints for her response but she is on vacation at the moment.
Her respond pending, the right design style that match our guidance and existing control would to do acrylic and do 1px border which matches the color that we use on controls like MenuFlyout.

**Comment by Felix-Dev** [comment]
@chigy Thanks for the update. I will wait for @kikisaints then to give her view on this.

**Comment by kikisaints** [comment]
I believe less than 32x32 on both axes was the issue. If this new suggestion is bigger than 32 on either axis, it should be fine.

It's also not that we can't do it, it's more that something that small is spending a lot of cycles/performance to render something that's virtually undetectable. So if we feel this acrylic surface (which is in-line with the guidance by being a popup) is actually noticeable as an acrylic surface than I'm all for it.

**Comment by Felix-Dev** [comment]
@kikisaints The popup spin buttons are currently of size 40x32 (WidthxHeight): https://github.com/microsoft/microsoft-ui-xaml/blob/9b5509d02d1c8d074dbe6b0705d5893bfa87a7d4/dev/NumberBox/NumberBox.xaml#L194

I will hopefully be able today to create another screenshot showing how the design will look with the guidance given by @chigy.

**Comment by Felix-Dev** [comment]
> Her respond pending, the right design style that match our guidance and existing control would to do acrylic and do 1px border which matches the color that we use on controls like MenuFlyout.

@kikisaints @chigy For demo purposes *only* I modified the NumberBox to show three SpinButton popups next to each other. The `MenuFlyout` and `ComboBox` styles use the `SystemControlTransientBackgroundBrush` theme resource for their transient UIs. The `CommandBarFlyout` style uses the `SystemControlAcrylicElementBrush`. So we have three transient UIs but the CommandBarFlyout uses a different background for its transient UI.

Below I've compared both brushes and how they would look like for the NumberBox spin buttons popup:

| Theme | SystemControlTransientBackgroundBrush | SystemControlAcrylicElementBrush |
|----|----|----|
| Light | ![image](https://user-images.githubusercontent.com/1398851/90244288-c0a42200-de30-11ea-8e43-7845a166d7b5.png)| ![image](https://user-images.githubusercontent.com/1398851/90244252-ad915200-de30-11ea-9e4c-a8b4f0dd402f.png)|
| Dark | ![image](https://user-images.githubusercontent.com/1398851/90244306-cbf74d80-de30-11ea-989f-6b8f2e1cf606.png)| ![image](https://user-images.githubusercontent.com/1398851/90244333-d44f8880-de30-11ea-9c0c-f973d8408000.png)|

As you can hopefully see there is a visual difference here depending on which type of brush I am using. Which one do you prefer?

**Comment by mdtauk** [comment]
@Felix-Dev the border in the Dark Theme, could do with being lighter, to help the buttons pop from the dark background

**Comment by Felix-Dev** [comment]
@mdtauk The border I am currently using here is the same one used for the MenuFlyout and CommandBarFlyout - `SystemControlTransientBorderBrush`:

![image](https://user-images.githubusercontent.com/1398851/90253348-4b8d1880-de41-11ea-926f-8f65bff70417.png)
(A CommandBarFlyout shown on the default page background in dark theme)

As the `SystemControlTransientBorderBrush` is slightly visible in Light theme bout not in Dark theme, we could update that theme resource for the dark theme to make the border slightly visible here as well. The new borderbrush would then be reflected across the board for the different popups and flyouts used in WinUI. 

@chigy @kikisaints FYI

**Comment by mdtauk** [comment]
Maybe the top edge highlight should be used for these controls, as with 10X

![image](https://user-images.githubusercontent.com/7389110/90255351-1500cf00-de3c-11ea-8281-8d5c37e02494.png)

**Comment by chigy** [comment]
@mdtauk , @Felix-Dev , 
We use the same brush for the borders across all of our controls. Please stick with the same value.

**Comment by mdtauk** [comment]
> @mdtauk , @Felix-Dev ,
> We use the same brush for the borders across all of our controls. Please stick with the same value.

That makes sense, however it is still harder to see the Popup Buttons in the Dark Theme, as it uses a dark border and not a lighter one.

**Comment by Felix-Dev** [comment]
@chigy I believe @mdtauk's suggestion here is to look at the borderbrush used in dark theme (SystemControlTransientBorderBrush) for transient controls and modify it slighty for better visibility. For example, see the color of the `SystemControlTransientBorderBrush` in Light theme: You can clearly see the border of the transient control. That's not the case in dark theme for now. The suggestion here is to look at the `SystemControlTransientBorderBrush` value in dark mode and perhaps update it for better border visibility.

Modifying this brush would not break consistency within WinUI as I believe all transient UIs use that theme resource as their borderbrush.

**Comment by chigy** [comment]
@mdtauk , @Felix-Dev ,
What I am asking is please don't design new behavior for this control. If you are suggesting to update the border, then please open a separate new GitHub item and address it for all controls. We do not want to introduce one-off.

**Comment by Felix-Dev** [comment]
@chigy It seems you have missunderstood me. I was not planning to update the border *just* for the NumberBox control here.

**Comment by mdtauk** [comment]
![image](https://user-images.githubusercontent.com/7389110/90265235-2e107c80-de4a-11ea-94ce-9e048ce2fac8.png)
![image](https://user-images.githubusercontent.com/7389110/90265276-3ff21f80-de4a-11ea-9433-76e6bfccdcb1.png)

It seems to me that if the border is there to improve visibility, the dark theme should use a lighter colour

![image](https://user-images.githubusercontent.com/7389110/90265384-6adc7380-de4a-11ea-97d1-654e8893ef44.png)

**Comment by Felix-Dev** [comment]
@mdtauk Can we move updating the borderbrush for transient UIs in dark mode into its separate issue? It's not a NumberBox spin buttons popup specific issue (though if you plan to open a tracking issue for this you can reference the NumberBox as an example).

**Comment by mdtauk** [comment]
Feel free to use the images when setting up a new issue.

**Comment by chigy** [comment]
> @chigy It seems you have missunderstood me. I was not planning to update the border _just_ for the NumberBox control here.

@Felix-Dev , I understood you perfectly.
I am asking, please do not introduce a change that impacts all of the controls through an issue that is specifically about NumberBox. Please open a separate one and we will discuss it seprately.

**Comment by Felix-Dev** [comment]
@chigy Ok, then it seems i have missunderstood you :)

Anyway, rest assured, I wasn't thinking about doing that here since that would be an update much broader in scope than just the NumberBox. This will be tracked in its own issue which with @mdtauk's consent I would open later today or tomorrow.

Back to the NumberBox specific issue here: Do you have an opinion on whether I should use the `SystemControlTransientBorderBrush` theme resource or the `SystemControlAcrylicElementBrush` theme resource?

**Comment by chigy** [comment]
@Felix-Dev , I don't know the specific brushes used but please match the color exactly with the MenuFlyout.
![image](https://user-images.githubusercontent.com/22958680/90267242-216c3400-de0a-11ea-8add-2c6c040a68c0.png)

**Comment by Felix-Dev** [comment]
@chigy Alright, will do.
